src/query/block/scalar.go

```diff
@@ -138,12 +138,78 @@ type scalarStep struct {
 func (it *scalarStep) Time() time.Time   { return it.time }
 func (it *scalarStep) Values() []float64 { return it.vals }
 
-// SeriesIter is invalid for a scalar block.
+// SeriesIter returns an iter containing the given value across a single series.
 func (b *Scalar) SeriesIter() (SeriesIter, error) {
-	return nil, errors.New("series iterator undefined for a scalar block")
+	return &scalarSeriesIter{
+		idx:  -1,
+		val:  b.val,
+		meta: b.meta,
+	}, nil
```

Comment:
nit: Same comment here about using a `newScalarSeriesIter(...)`. Take it or leave it 👍